src/ws/cockpit-session-client-certificate.c

```diff
@@ -138,69 +153,79 @@ https_instance_has_certificate_file (char   *contents,
   if (https_instance == NULL) /* already warned */
     goto out;
 
+  /* No tricky stuff, please */
+  if (!valid_256_bit_hex_string (filename))
+    {
+      warnx ("tls-cert authentication token is invalid");
+      goto out;
+    }
```

Comment:
I wonder if that extra effort with the helper function is really justified. The check immediately below already validates that it has the only possible value -- so any "funny" stuff would be caught there anyway.

Reply:
But this changes in future commits...